stdlib/http/src/main/java/org/ballerinalang/net/http/nativeimpl/connection/Respond.java

```diff
@@ -108,14 +113,30 @@ public static Object nativeRespond(ObjectValue connectionObj, ObjectValue outbou
             }
         } catch (ErrorValue e) {
             unBlockStrand(strand);
-            log.debug(e.getPrintableStackTrace(), e);
-            return e;
+            if (log.isDebugEnabled()) {
+                log.debug(e.getPrintableStackTrace(), e);
+            }
+            if (connectionObj.getNativeData(WebSocketConstants.WEBSOCKET_HANDSHAKER) != null) {
+                throw getWebSocketException(outboundResponseObj);
+            } else {
+                return e;
```

Comment:
Is there a specific use case when `connectionObj.getNativeData(WebSocketConstants.WEBSOCKET_HANDSHAKER)` can be null?

Reply:
Yes. That is indicated this is not a request of the http Websocket upgrade.